### PR TITLE
feat(route53): adding ipv6 option for Route53 Alias

### PR DESCRIPTION
**Pull request by makrsmark**
----

### Pull Request Checklist

* [x] Testing
  - Unit test added
  - __CLI change?:__ manually run [integration tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh) and paste output as a PR comment
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-route53/README.md [hunk 1]
```diff
@@ -28,6 +28,25 @@ const zone = new route53.PrivateHostedZone(this, 'HostedZone', {
...
 Additional VPCs can be added with `zone.addVPC()`.
 
+### Adding Alias
+
+To add an alias to your zone:
+```ts
+import route53 = require('@aws-cdk/aws-route53');
+
+// For the example, using existing hosted zone
+const _baz = route53.HostedZone.import(this, 'baz', {
```
Comment (sam-goodwin) [inline packages/@aws-cdk/aws-route53/README.md]: I don't think `_baz` is consistent with our naming conventions. Change to `baz`?

File: packages/@aws-cdk/aws-route53/test/test.alias-record.ts [hunk 2]
```diff
@@ -38,6 +38,80 @@ export = {
...
+          hostedZoneId: 'Z2P70J7EXAMPLE',
+          dnsName: 'foo.example.com'
+        };
+      }
+    };
+
+    // WHEN
+    new AliasRecord(zone, 'Alias', {
+      zone,
+      recordName: '_foo',
```
Comment (sam-goodwin) [inline packages/@aws-cdk/aws-route53/test/test.alias-record.ts]: `foo`

Comment (makrsmark) [reply packages/@aws-cdk/aws-route53/test/test.alias-record.ts]: will change

File: packages/@aws-cdk/aws-route53/README.md (hunk 1 above)
Comment (makrsmark) [reply packages/@aws-cdk/aws-route53/README.md]: will change.

File: packages/@aws-cdk/aws-route53/lib/records/alias.ts [hunk 3]
```diff
@@ -43,6 +42,10 @@ export interface AliasRecordProps {
    * Target for the alias record
    */
   target: IAliasRecordTarget;
+  /**
+   * The type of the alias (A or AAAA). Defaults to 'A'.
```
[the remark this answers is not in the document]
Comment (makrsmark) [reply packages/@aws-cdk/aws-route53/lib/records/alias.ts]: I thought about it, but didn't since existing code doesn't enumerate. I'll add and update.

**Comment by eladb** [comment]
@makrsmark closing for now. feel free to reopen when/if you wish to continue this work

**Pull request closed without merging**